fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/AggregateFunction.java

```diff
@@ -17,31 +17,59 @@
 
 package org.apache.doris.nereids.trees.expressions.functions;
 
+import org.apache.doris.nereids.exceptions.UnboundException;
 import org.apache.doris.nereids.trees.expressions.Expression;
 import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
 import org.apache.doris.nereids.types.DataType;
 
 /** AggregateFunction. */
 public abstract class AggregateFunction extends BoundFunction {
 
-    private DataType intermediate;
+    /**
+     * indicate current function used in local stage or global stage.
+     */
+    protected final boolean isLocal;
 
     public AggregateFunction(String name, Expression... arguments) {
+        this(name, false, arguments);
+    }
+
+    public AggregateFunction(String name, boolean isLocal, Expression... arguments) {
         super(name, arguments);
+        this.isLocal = isLocal;
+    }
+
+    public boolean isLocal() {
+        return isLocal;
     }
 
     public abstract DataType getIntermediateType();
 
+    /**
+     * When do two stage aggregate, the local stage's output type is different from the global.
+     * This method is used to get the local stage's output type.
+     */
+    public abstract DataType getLocalDataType();
```

Comment:
Why don't return the getDataType refer to the isLocal property, but use two distinct funciton to do this?